### PR TITLE
PP-10439 Only create agreements when RCP enabled

### PR DESCRIPTION
**Pull request by james-peacock-ho**
- It should only be possible to create agreements when a gateway account has RCP enabled (recurring_enabled = true)
- If account does not have RCP enabled, throw custom exception
- Respond with 422 with error identifier RECURRING_CARD_PAYMENTS_NOT_ALLOWED
- Add unit test for this scenario and update existing
- PACT tests to be added under a separate ticket PP-10443

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/agreement/exception/RecurringCardPaymentsNotAllowedExceptionMapper.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+public class RecurringCardPaymentsNotAllowedExceptionMapper 
+        implements ExceptionMapper<RecurringCardPaymentsNotAllowedException> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RecurringCardPaymentsNotAllowedExceptionMapper.class);
+
+    private static final String RESPONSE_ERROR_MESSAGE = "Recurring payment agreements are not enabled on this account";
+
+    @Override
+    public Response toResponse(RecurringCardPaymentsNotAllowedException exception) {
```
Comment (kbottla) [inline src/main/java/uk/gov/pay/connector/agreement/exception/RecurringCardPaymentsNotAllowedExceptionMapper.java:20]: Can we have an integration test for this error? ITest makes sure that the mapper is configured correctly and that the response is as expected.

Comment (james-peacock-ho) [reply src/main/java/uk/gov/pay/connector/agreement/exception/RecurringCardPaymentsNotAllowedExceptionMapper.java:20]: test added. thanks!

File: src/test/java/uk/gov/pay/connector/agreement/resource/AgreementsApiResourceIT.java [hunk 2]
```diff
@@ -154,6 +154,21 @@ public void shouldReturn422WhenDescriptionEmpty() throws JsonProcessingException
...
+        String payload = objectMapper.writeValueAsString(Map.of(
+                "reference", REFERENCE_ID,
+                "description", DESCRIPTION,
+                "user_identifier", USER_IDENTIFIER
+        ));
+        givenSetup()
+                .body(payload)
+                .post(format(CREATE_AGREEMENT_URL, accountId))
+                .then()
+                .statusCode(SC_UNPROCESSABLE_ENTITY);
```
Comment (kbottla) [inline src/test/java/uk/gov/pay/connector/agreement/resource/AgreementsApiResourceIT.java]: Can the error identifier `RECURRING_CARD_PAYMENTS_NOT_ALLOWED ` and the message checked too?

Comment (james-peacock-ho) [reply src/test/java/uk/gov/pay/connector/agreement/resource/AgreementsApiResourceIT.java]: thanks - that's updated now

**Review by kbottla — APPROVED** [review]

**Pull request merged**